src/analyses/README.md

```diff
@@ -92,7 +92,62 @@ To be documented.
 
 \subsection analyses-dependence-graph Data- and control-dependence analysis (dependence_grapht)
 
-To be documented.
+### Dependence graph
+
+Implemented in `src/analyses/dependence_graph.h(cpp)`. It is a graph and an
+abstract interpreter in the same time. The abstract interpretation nature
+allows a dependence graph to build itself (the graph) from a given GOTO program.
```

Comment:
what does it mean for a graph to build itself?

Reply:
For a graph in general it would perhaps mean nothing. However, here we speak about a special graph - the dependence graph in our code base. It is not just a graph representing control/data-dependencies is some GOTO program. It is also AI analyser. You do not build it by calling its graph-interface methods for adding nodes and edges. You instead pass it GOTO program and it will run its AI-interface methods on it to perform the data- and control- dependency analyses (i.e. it uses its "AI" nature of being also an AI analyser) and then it runs its `finalize` method to actually build the graph by calling graph-interface methods (to add control/data-dependency edges) from the results of the analyses.

Reply:
I think the wording here ("build itself") is just a bit weird, below you very nicely explain what is going on and it may be worth having a forward reference to that.